ospfd/ospfd.c

```diff
@@ -380,18 +414,50 @@ struct ospf *ospf_get_instance(u_short instance)
 
 	ospf = ospf_lookup_instance(instance);
 	if (ospf == NULL) {
-		ospf = ospf_new(instance);
+		ospf = ospf_new(instance, NULL /* VRF_DEFAULT*/);
 		ospf_add(ospf);
 
-		if (ospf->router_id_static.s_addr == 0)
+		if (ospf->router_id_static.s_addr == 0) {
+			if (vrf_lookup_by_id(ospf->vrf_id))
+				ospf_router_id_update(ospf);
+			else {
+				if (IS_DEBUG_OSPF_EVENT)
+					zlog_debug("%s: ospf VRF (id %d) is not active yet, skip router id update"
+						    , __PRETTY_FUNCTION__,
+						    ospf->vrf_id);
+			}
 			ospf_router_id_update(ospf);
+		}
 
 		ospf_opaque_type11_lsa_init(ospf);
 	}
 
 	return ospf;
 }
 
+struct ospf *ospf_lookup_by_vrf_id(vrf_id_t vrf_id)
```

Comment:
i would probably rename this to `ospf_lookup`

Reply:
Would it be ok to keep vrf id based name vs. legacy api name? Also checked bgpd where by_vrf_id name of the api used. Like to keep symmetric.